### PR TITLE
Add thread

### PR DESCRIPTION
**Pull request by MiLk**
Will replace #28.

Waiting for https://github.com/cinemast/libjson-rpc-cpp/pull/56.

### CODE REVIEW

**Comment by MiLk** [comment]
Ready to merge.

Please check that you can run `./examples/examples_callback_thread`.

File: vendor/libjson-rpc-cpp [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit a34172ce664abdcab107ca0cc71c83d0cd50b202
+Subproject commit 0a366be7b8b8e727b052164bc05783935df42a79
```
Comment (ronkorving) [inline vendor/libjson-rpc-cpp:1]: Is this update intentional?

Comment (MiLk) [inline vendor/libjson-rpc-cpp:1]: Look at the commit message ;) https://github.com/MiLk/mage-sdk-cpp/commit/d9049d3a150b1814fa1cb87b3253e82983f1c826

It's for https://github.com/cinemast/libjson-rpc-cpp/pull/56

Comment (ronkorving) [inline vendor/libjson-rpc-cpp:1]: Okay, cool :) Just wanted to be sure (but was also a bit lazy I guess), as I've seen many PRs accidentically forcing submodules to commit hashes that were not intended to happen.

**Pull request merged**